File: calculator.py
import webbrowser
from threading import Timer
from flask import Flask, request, jsonify
from flask_cors import CORS
import os

app = Flask(__name__)
CORS(app)


@app.route('/api/calculate', methods=['POST'])
def calculate():
    data = request.json
    a, b, op = float(data['a']), float(data['b']), data['operator']
    res = 0
    if op == '+': res = a + b
    elif op == '-': res = a - b
    elif op == '*': res = a * b
    elif op == '/': res = "Hiba" if b == 0 else a / b
    return jsonify({"result": res})

@app.route('/api/ai', methods=['POST'])
    
def open_browser():
    path = os.path.abspath("index.html")
    webbrowser.open(f"file://{path}")

if __name__ == '__main__':
    Timer(1.5, open_browser).start()
    app.run(port=5000)

File: backend_rust/src/calculator.rs
pub fn calculate(a: f64, b: f64, operator: &str) -> String {
    match operator {
        "+" => (a + b).to_string(),
        "-" => (a - b).to_string(),
        "*" => (a * b).to_string(),
        "/" => if b == 0.0 { "Hiba".to_string() } else { (a / b).to_string() },
        _ => "Hiba".to_string(),
    }
}